projects/solar/src/sense_mppt.c

```diff
@@ -0,0 +1,76 @@
+#include "sense_mppt.h"
+
+#include "data_store.h"
+#include "event_queue.h"
+#include "log.h"
+#include "mppt.h"
+#include "sense.h"
+#include "solar_boards.h"
+#include "solar_events.h"
+#include "spv1020_mppt.h"
+#include "status.h"
+
+static SpiPort s_spi_port;
+static uint8_t s_indices[MAX_SOLAR_BOARD_MPPTS];
+
+static void prv_check_status_for_faults(uint8_t mppt, uint8_t status) {
+  uint8_t ovc_branch_bitmask;
+  if (spv1020_is_overcurrent(status, &ovc_branch_bitmask)) {
+    uint16_t ovc_data = (ovc_branch_bitmask << 8) | mppt;
+    event_raise(SOLAR_FAULT_EVENT_MPPT_OVERCURRENT, ovc_data);
+  }
+  if (spv1020_is_overvoltage(status)) {
+    event_raise(SOLAR_FAULT_EVENT_MPPT_OVERVOLTAGE, mppt);
+  }
+  if (spv1020_is_overtemperature(status)) {
+    event_raise(SOLAR_FAULT_EVENT_MPPT_OVERTEMPERATURE, mppt);
+  }
+}
+
+static void prv_sense_cycle_callback(void *context) {
+  uint8_t mppt = *(uint8_t *)context;
```

Comment:
do you think it makes sense to `typedef uint8_t Mppt`?

Reply:
I think that makes sense.